src/components/YourCompany/OfficeAddresses.vue

```diff
@@ -282,171 +315,164 @@ export default class OfficeAddresses extends Mixins(CommonMixin, EntityFilterMix
-      }
+  /** True if (registered) delivery address has changed. */
+  private get deliveryChanged (): boolean {
+    return !this.isSame(this.getOfficeAddresses.registeredOffice?.deliveryAddress,
+      this.originalOfficeAddresses.registeredOffice?.deliveryAddress)
+  }
 
-      // Records Delivery Address
-      if (this.$refs.recMailingAddress?.$refs?.addressForm && !this.inheritRecMailingAddress &&
-          !this.isSame(this.recDeliveryAddress, this.defaultAddress)) {
-        this.$refs.recDeliveryAddress.$refs.addressForm.validate()
-      }
-    }
+  /** True if records mailing address has changed. */
+  private get recMailingChanged (): boolean {
+    return !this.isSame(this.getOfficeAddresses.recordsOffice?.mailingAddress,
+      this.originalOfficeAddresses.recordsOffice?.mailingAddress)
   }
 
-  /**
-   * Sets address data.
-   * @param loadInheritedFlags used to update inherited flags based on isSame checks if true
-   */
-  private initializeAddresses (loadInheritedFlags: boolean): void {
-    if (this.addresses.registeredOffice) {
-      this.mailingAddress = this.addresses.registeredOffice.mailingAddress
-      this.deliveryAddress = this.addresses.registeredOffice.deliveryAddress
-
-      if (loadInheritedFlags) {
-        this.inheritMailingAddress = this.isSame(
-          this.addresses.registeredOffice.mailingAddress,
-          this.addresses.registeredOffice.deliveryAddress
-        )
-      }
-      if (this.entityFilter(EntityTypes.BCOMP)) {
-        this.recMailingAddress = this.addresses.recordsOffice?.mailingAddress
-        this.recDeliveryAddress = this.addresses.recordsOffice?.deliveryAddress
-
-        if (loadInheritedFlags) {
-          this.inheritRegisteredAddress = (
-            this.isSame(
-              this.addresses.registeredOffice.deliveryAddress,
-              this.addresses.recordsOffice?.deliveryAddress
-            ) && this.isSame(
-              this.addresses.registeredOffice.mailingAddress,
-              this.addresses.recordsOffice?.mailingAddress
-            )
-          )
+  /** True if records delivery address has changed. */
+  private get recDeliveryChanged (): boolean {
+    return !this.isSame(this.getOfficeAddresses.recordsOffice?.deliveryAddress,
+      this.originalOfficeAddresses.recordsOffice?.deliveryAddress)
+  }
 
-          this.inheritRecMailingAddress = this.isSame(
-            this.addresses.recordsOffice?.mailingAddress,
-            this.addresses.recordsOffice?.deliveryAddress
-          )
-        }
-      }
-    }
+  /** True if any office address has changed. */
+  private get officeAddressesChanged (): boolean {
+    return (this.mailingChanged || this.deliveryChanged || this.recMailingChanged || this.recDeliveryChanged)
   }
 
-  //
-  // Getters (Computed Values)
-  //
-  /** Whether the address form is valid. */
+  /** True if the address form is valid. */
   private get formValid (): boolean {
```

Comment:
Yes, I love getters. Did you know that they cache their value if it hasn't changed (ie, if what it depends on hasn't changed)?

Reply:
i see the light on getters. They are definitely useful!

Reply:
And they're fast. Have you ever seen a page render slowly? That's possibly because it re-evaluates some code conditions many times during render. In such cases, a getter can greatly improve performance.

I'm impressed by Vue getters actually, as they can be quite complex and even depend on other getters.